### ISSUE TITLE
INSERT INTO FUNCTION cluster IN JOIN table

### ISSUE THREAD

**Issue by sablastik** [ClickHouse/ClickHouse#32963]
Clickhouse 21.8.12.29.
Cluster of 2 shards with 2 replicas

I'm trying to INSERT data into table with ENGING=Join. My aim is to get similar data on all replicas.
Data INSERTS as SELECT from table with ENGINE=Distributed over ReplicatedMergeTree:
```
INSERT INTO FUNCTION cluster('my_clusters', 'tmp', 'test_table')
SELECT column1, column2
FROM db.table_with_distributed_engine; 
```
As far as I'm not setting sharding key (I don't need join table to be sharded) I'm getting error:
`Code: 55, e.displayText() = DB::Exception: Method write is not supported by storage Distributed with more than one shard and no sharding key provided (version 21.8.12.29 (official build))`

But I'm expecting to get full copied of data on all shards.

**Comment by den-crane** [comment]
Please provide a `my_clusters` xml config.

**Comment by den-crane** [comment]
the pretty same question https://github.com/ClickHouse/ClickHouse/issues/32995

**Comment by sablastik** [comment]
> Please provide a `my_clusters` xml config.

<remote_servers>
        <start>
            <shard>
                <weight>10</weight>
                <internal_replication>true</internal_replication>
                <replica>
                    <host>c1</host>
                    <port>9000</port>
                    <user>login</user>
                    <password>pass</password>
                </replica>
                <replica>
                    <host>c2</host>
                    <port>9000</port>
                    <user>login</user>
                    <password>pass</password>
                </replica>
            </shard>
            <shard>
                <weight>10</weight>
                <internal_replication>true</internal_replication>
                <replica>
                    <host>c3</host>
                    <port>9000</port>
                    <user>login</user>
                    <password>pass</password>
                </replica>
                <replica>
                    <host>c4</host>
                    <port>9000</port>
                    <user>login</user>
                    <password>pass</password>
                </replica>
            </shard>
        </start>
    </remote_servers>

**Comment by sablastik** [comment]
> the pretty same question #32995

Not exactly. I know how to get exactly the same data in tables on all replicas. However I'm just trying to understand why 
`INSERT INTO FUNCTION cluster(` not working.

I'm expecting that this function should do the same as if i wrote 4 queries like:
```
INSERT INTO FUNCTION remote('c1', 'tmp', 'test_table') VALUES ('test', 42) SELECT column1, column2
FROM db.table_with_distributed_engine;
INSERT INTO FUNCTION remote('c2', 'tmp', 'test_table') VALUES ('test', 42) SELECT column1, column2
FROM db.table_with_distributed_engine;
INSERT INTO FUNCTION remote('c3', 'tmp', 'test_table') VALUES ('test', 42) SELECT column1, column2
FROM db.table_with_distributed_engine;
INSERT INTO FUNCTION remote('c4', 'tmp', 'test_table') VALUES ('test', 42) SELECT column1, column2
FROM db.table_with_distributed_engine;
```
Thats why i've tagged it as Unexpected behavior.

**Comment by den-crane** [comment]
You have 2 shards. What do you expect?

**Comment by sablastik** [comment]
I expect that query 
```
SELECT column1, column2
FROM db.table_with_distributed_engine; 
``` 
would be executed on the initial server and the query result would be inserted into all 4 instances.

**Comment by sablastik** [comment]
I've found more suitable example to explain what I expect.

In my opinion `INSERT INTO FUNCTION cluster()` should have the same behavior as if I do 
```
CREATE TABLE tmp.test_table ENGINE=Join(ANY, LEFT, key_column) AS
SELECT column1, column2
FROM db.table_with_distributed_engine; 
```

**Comment by inertance** [comment]
how can i config this in jdbc, when i use datax ,write data into clickhouse distributed table ,it also occur
